### DISCUSSION TITLE
[how-to] Basic integration with Contact Form 7 - Wordpress & YetiREST API

### DISCUSSION THREAD

**Discussion by apcloic** [YetiForceCompany/YetiForceCRM#13932] [How to]
Hi,

I've just finished a simple integration of YetiREST API in Wordpress so I would like to share it with YF community.

Requirements :
- Wordpress site with Contact Form 7 plugin.

Use case : when someone fill the form on your wordpress website, a new contact record is created with its filled name and email.
Bear in mind that this is really basic (no error check, etc.).

1- Configure and activate API on Yetiforce.
#11770

2- Download a copy of YetiREST
https://github.com/YetiForceCompany/YetiForceScripts/tree/master/YetiREST

3 - Put it in a Wordpress folder (for example in /wp-content/themes/your-child-theme/includes)
**Note :  Don't forget to install dependencies with composer**

4 - In YetiREST/config.php replace default values with your own (created on yetiforce side in webservice - applications and webservice - users modules)

```
return [
	'apiPath' => 'https://yetiforce.url/',
	'wsAppName' => 'replace_with_your_webservice_app_name',
	'wsAppPass' => 'replace_with_your_webservice_app_password',
	'wsApiKey' => 'replace_with_your_webservice_app_key',
	'wsUserName' => 'replace_with_your_webservice_user_login',
	'wsUserPass' => 'replace_with_your_webservice_user_password'
];

```

5 - In the file functions.php, add and adapt the following code :

```

// Include YetiREST API
require_once( __DIR__ . '/includes/YetiREST/vendor/autoload.php'); // you may have to change the path depending on where you decide to put YetiREST folder

function on_submit( $form, &$abort, $submission )
{

// Select which contact form 7 should load the function

	if ( $abort === TRUE || $form->ID() !== 'CONTACT_FORM7_ID' ) {  // Replace with your own CONTACT_FORM7_ID
	   return;
	}

	// Get form data

	$data = $submission->get_posted_data();

	$email = sanitize_text_field($data['your-email']); // Where your-email is the name of your form field, change if needed
	$name = sanitize_text_field($data['your-name']); // Where your-name is the name of your form field

	// 'lastname' (required) and 'email' (optional) are the field's name on destination module in yetifore, add other other required fields if needed.

  	$recorddata = array(
    		'lastname' => $name,
    		'email' => $email
  	);

	$moduleName = 'Contacts'; // Could be replaced with Accounts, Leads, ...

	// create new record in Yetiforce CRM

	$api = new \App\Portal();
	$api->debug = true;
	$api->init();

	if ($login = $api->login()) {

	$response = $api->json('POST', $moduleName."/Record", $recorddata);

	}
   
}

add_action('wpcf7_before_send_mail', 'on_submit', 10, 3);

```

As I'm not a programer, let me know if there is a better way to use Yetiforce REST API with Wordpress.

Build with the help of the following page : https://dev.to/jorbascrumps/integrating-apis-with-wordpress-and-contact-form-7-1657

Regards,

**Comment by simple-AMS** [comment]
Wow! That's great work @apcloic and an amazing guide you shared. This takes me back to the Webform days where a potential lead filled out a form and boom instantly you had a newly created lead in the system. 

I will certainly be giving this a shot soon and will let you know the end result. Thanks again man, if this works you're my hero!
